h/streamer/filter.py

```diff
@@ -28,19 +28,29 @@ class SocketFilter:
     KNOWN_FIELDS = {"/id", "/uri", "/references"}
 
     @classmethod
-    def matching(cls, sockets, annotation):
+    def matching(cls, sockets, annotation, equivalent_uris=None):
         """Find sockets with matching filters for the given annotation.
 
         For this to work, the sockets must have first had `set_filter()` called
         on them.
 
         :param sockets: Iterable of sockets to check
         :param annotation: Annotation to match
+        :param equivalent_uris: URIs equivalent to this annotation's `target_uri`.
+                                These can be looked up using `h.storage.expand_uri`.
+        :type equivalent_uris: Iterable[str]
         :return: A generator of matching socket objects
         """
+
+        # The normalization of URIs here prior to matching against filters should
+        # match the normalization done when searching (see `h.search.query`).
+        uris = {normalize_uri(annotation.target_uri)}
+        if equivalent_uris:
+            uris.update(normalize_uri(u) for u in equivalent_uris)
```

Comment:
~Is it necessary to normalise the return values here given that they came from `expand_uri`. Shouldn't they already be normalised?~ They actually aren't but they easily could be. Currently `expand_uri` returns `docuri.uri`, but another column on the same table is `uri_normalized`. So it wouldn't be hard to adapt that method to accept a `normalized=True` method and skip all runtime normalisation processing. Hopefully this would be faster.

Also a small thing, but when you call `expand_uri` presumably one of the returned items is the normalised version of the original URL? It seems we could skip adding the normalised `target_uri` here as `expand_uri` will do it for us? If true this would save a little time.

Along with the suggestion to move the normalisation here, I think this could simplify this section to:

```suggestion
        uris = set(storage.expand_uri(session, annotation.target_uri, normalize=True))
```